test/test_http.py

```diff
@@ -148,6 +150,31 @@ def do_GET(self):
             self.send_header('Location', new_url)
             self.send_header('Content-Length', '0')
             self.end_headers()
+        elif self.path == '/content-encoding':
+            encodings = self.headers.get('ytdl-encoding', '').split(',')
+            payload = b'<html><video src="/vid.mp4" /></html>'
+            for encoding in encodings:
```

Comment:
It seems strange to `str.split` here and then `",".join` them later down. 
```suggestion
            encodings = self.headers.get('ytdl-encoding')
            payload = b'<html><video src="/vid.mp4" /></html>'
            for encoding in filter(None, (encodings or '').split(',')):
```
```diff
-                elif encoding:
+                else:
                    self._status(415)
                    return
```
```diff
-            self.send_header('Content-Encoding', ','.join(encodings))
+            if encodings:
+                self.send_header('Content-Encoding', encodings)
```

Reply:
needs to be `elif encoding` to only catch if `encoding` is set. if it isn't then no need to encode anything

Reply:
Thats where the `filter(None, ...)` came into play. I thought its clearer filtering before as to not do the loop at all if encoding is not set.

Reply:
oh true true